sympy/utilities/lambdify.py

```diff
@@ -179,7 +179,7 @@ def _import(module, reload=False):
 
 
 @doctest_depends_on(modules=('numpy', 'scipy', 'tensorflow',), python_version=(3,))
-def lambdify(args: tUnion[Iterable, 'sympy.core.expr.Expr'], expr: 'sympy.core.expr.Expr', modules=None, printer=None, use_imps=True,
```

Comment:
If we did have a type hint here it would be better to write it more concisely as `Iterable | Expr`. 

However, I think this is an example of something I touched on in https://github.com/sympy/sympy/issues/17945. The hint here is not actually correct. The `args` can be an arbitrarily nested iterable, like `((x, y), z)` (it also really should be Sequence, not Iterable, since unordered iterables like `set` are not allowed). To make nested sequences work, you need something like this https://github.com/numpy/numpy/pull/20129.

Reply:
Yes, I noticed it wasn't correct either. It is just confusing, is what it is.